Posts/2021/07/How Can I Be Notified Any Time a Service Goes Down.md

```diff
@@ -0,0 +1,186 @@
+---
+post_title: How Can I Be Notified Any Time a Service Goes Down?
+username: farisnt@gmail.com
+Categories: PowerShell
+tags: PowerShell, WMI, Events
+Summary: Using Powershell to create temporary event monitoring using WMI
+
+---
+
+**How Can I Be Notified Any Time a Service Goes Down?**
+
+Q: How Can I Be Notified Any Time a Service Goes Down?
+
+A: The short quick answer to utilizing WMI and PowerShell 7.
+
+You use Powershell to create temporary event monitoring using WMI.
+Then WMI monitors any service changes and generates an alert once it detects a change.
+
+## Basic Requirement
+
+To achieve this, you need Windows PowerShell 5.1 and above.
+
+This post uses the latest version of PowerShell 7.
+So if you are don't yet have PowerShell 7, see the [Microsoft documentation on how to install PowerShell 7 on Windows.](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-7.1).
+
+Also, make sure that PowerShell is running as administrator.
+
+## Finding the Required Class
+
+Before going into details, you need to find the required class to monitor.
+To get a list of all available classes, use the following code.
+
+```powershell-console
+PS> Get-CimClass -Namespace root\cimv2

```

Comment:
Must use double backslash. This is a requirement imposed by the process that copies the post from GitHub to WordPress.

```suggestion
PS> Get-CimClass -Namespace root\\cimv2
```